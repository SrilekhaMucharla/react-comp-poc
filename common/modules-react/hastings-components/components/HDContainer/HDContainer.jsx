import PropTypes from 'prop-types';
import React from 'react';
import { Col } from 'react-bootstrap';
import HDLabelRefactor from '../HDLabelRefactor/HDLabelRefactor';
import HDTickListItem from '../HDTickListItem/HDTickListItem';
import * as messages from './HDContainer.messages';

const HDContainer = ({
    hastingsDirectLogo,
    isComprehensive,
    title, items,
    itemsNotCovered,
    isOnlineProductType
}) => {
    const coverTypeLabel = (isComprehensive) ? messages.comprehensiveLabel : messages.tpftLabel;
    const labelClass = isOnlineProductType ? "--online": "--label" ;
    const _md = isOnlineProductType ? 12 : 6;
    const _xs = isOnlineProductType ? 12 : 6;
    return (
        <div>
            {
                hastingsDirectLogo ? (
                    <>
                        <div className="policy-details-cover__title">
                            <Col xs={_xs} md={_md} lg="auto" className="pl-0">
                                <img className="policy-details-cover__logo img-fluid" src={hastingsDirectLogo} alt="logo" />
                            </Col>
                            <Col xs={_xs} md={_md} lg={8} className={`policy-details-cover__title${labelClass} px-0`} >
                                <HDLabelRefactor Tag="h5" text={coverTypeLabel} className="mb-0" />
                            </Col>
                        </div>
                    </>
                ) : (<HDLabelRefactor Tag="h4" text={title} className="policy-details-cover__extra-title" />)
            }

            <div className="policy-details-cover__list">
                {items.map((item) => (
                    <HDTickListItem
                        key={item.label || item.name}
                        title={item.label || item.name}
                        selected
                    >
                        {item.description}
                    </HDTickListItem>
                ))}
                {itemsNotCovered && itemsNotCovered.map((item) => (
                    <HDTickListItem
                        key={item.label || item.name}
                        title={item.label || item.name}
                    >
                        {item.description}
                    </HDTickListItem>
                ))}
            </div>
        </div>
    );
};

HDContainer.propTypes = {
    items: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        description: PropTypes.string,
    })).isRequired,
    title: PropTypes.string,
    isComprehensive: PropTypes.bool,
    hastingsDirectLogo: PropTypes.string,
    itemsNotCovered: PropTypes.arrayOf(PropTypes.shape({
        name: PropTypes.string,
        description: PropTypes.string,
    })),
    isOnlineProductType: PropTypes.bool
};

HDContainer.defaultProps = {
    title: null,
    isComprehensive: true,
    hastingsDirectLogo: null,
    itemsNotCovered: null,
    isOnlineProductType: false
};

export default HDContainer;
