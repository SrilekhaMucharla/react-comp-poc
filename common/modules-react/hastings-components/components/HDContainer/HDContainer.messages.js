export const comprehensiveLabel = 'Comprehensive';
export const tpftLabel = 'Third party fire and theft';
