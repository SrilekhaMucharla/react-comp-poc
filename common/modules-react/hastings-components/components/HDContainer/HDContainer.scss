.policy-details-cover__title{
    display: flex;
    flex-wrap: wrap;
    justify-content: space-between;
    margin-bottom: 22px;
    align-items: center;
    @include media-breakpoint-up(md) {
        justify-content: initial;
    }

    h5 {
        padding-top: 1%;
    }
    
    &--label {
        @include media-breakpoint-up(xxl) {
            padding-left: 0;
        }

        @include media-breakpoint-down(sm) {
            display: flex;
            justify-content: flex-end;
        }
    }

    &--online {
        @include media-breakpoint-up(xxl) {
            padding-left: 0;
        }
    }
}

.policy-details-cover__extra-title{
    margin-top: 30px;
    margin-bottom: 20px;
}

.policy-details-cover__logo{
    margin-right: 10px;
    @media (min-width: 768px) {
        margin-right: 35px;
    }
}
