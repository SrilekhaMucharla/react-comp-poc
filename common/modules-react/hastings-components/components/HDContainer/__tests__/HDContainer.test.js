import React from 'react';
import { shallow } from 'enzyme';
import HDContainer from '../HDContainer';

const items = [
    { name: 'Motor Legal Expenses' },
    { name: 'Roadside Assistance Breakdown cover' },
    { name: 'Driving other cars', description: 'Third party cover for the policyholder only, see your Certificate of Insurance for more information.' },
    { name: 'Courtesy car', description: 'A small car will be provided while your car\'s being repaired by one of our approved repairers.' },
];

const itemsNotCovered = [
    { name: 'Windscreen or windows' },
    { name: 'Fire and theft damage' }
];

describe('<HDContainer />', () => {
    it('render component with logo', () => {
        const wrapper = shallow(<HDContainer
            hastingsDirectLogo="logourl"
            isComprehensive={false}
            items={items}
            itemsNotCovered={[]} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render component with logo and comprehensive', () => {
        const wrapper = shallow(<HDContainer
            hastingsDirectLogo="logourl"
            items={items}
            itemsNotCovered={itemsNotCovered} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render component with and verify item', () => {
        const wrapper = shallow(<HDContainer
            hastingsDirectLogo="logourl"
            isComprehensive={false}
            items={items}
            itemsNotCovered={[]} />);
        expect(wrapper.find('HDTickListItem[title="Courtesy car"]').exists()).toBe(true);
    });

    it('render component and verify items not covered - Fire and theft damage', () => {
        const wrapper = shallow(<HDContainer
            hastingsDirectLogo="logourl"
            isComprehensive={false}
            items={items}
            itemsNotCovered={itemsNotCovered} />);
        expect(wrapper.find('HDTickListItem[title="Fire and theft damage"]').exists()).toBe(true);
    });

    it('render component and verify items not covered - Fire and theft damage', () => {
        const wrapper = shallow(<HDContainer
            hastingsDirectLogo="logourl"
            items={items} />);
        expect(wrapper.find('HDTickListItem[title="Fire and theft damage"]').exists()).toBe(false);
    });

    it('render component with title', () => {
        const wrapper = shallow(<HDContainer
            title="title"
            items={items} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('render component withuot header', () => {
        const wrapper = shallow(<HDContainer
            title="title"
            items={items} />);
        expect(wrapper).toMatchSnapshot();
    });
});
