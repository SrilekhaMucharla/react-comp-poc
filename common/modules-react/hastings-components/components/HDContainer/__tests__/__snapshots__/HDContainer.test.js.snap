// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HDContainer /> render component with logo 1`] = `
<div>
  <div
    className="policy-details-cover__title"
  >
    <Col
      className="pl-0"
      lg="auto"
      md={6}
      xs={6}
    >
      <img
        alt="logo"
        className="policy-details-cover__logo img-fluid"
        src="logourl"
      />
    </Col>
    <Col
      className="policy-details-cover__title--label px-0"
      lg={8}
      md={6}
      xs={6}
    >
      <HDLabelRefactor
        Tag="h5"
        adjustImagePosition={true}
        className="mb-0"
        text="Third party fire and theft"
      />
    </Col>
  </div>
  <div
    className="policy-details-cover__list"
  >
    <HDTickListItem
      key="Motor Legal Expenses"
      selected={true}
      title="Motor Legal Expenses"
    />
    <HDTickListItem
      key="Roadside Assistance Breakdown cover"
      selected={true}
      title="Roadside Assistance Breakdown cover"
    />
    <HDTickListItem
      key="Driving other cars"
      selected={true}
      title="Driving other cars"
    >
      Third party cover for the policyholder only, see your Certificate of Insurance for more information.
    </HDTickListItem>
    <HDTickListItem
      key="Courtesy car"
      selected={true}
      title="Courtesy car"
    >
      A small car will be provided while your car's being repaired by one of our approved repairers.
    </HDTickListItem>
  </div>
</div>
`;

exports[`<HDContainer /> render component with logo and comprehensive 1`] = `
<div>
  <div
    className="policy-details-cover__title"
  >
    <Col
      className="pl-0"
      lg="auto"
      md={6}
      xs={6}
    >
      <img
        alt="logo"
        className="policy-details-cover__logo img-fluid"
        src="logourl"
      />
    </Col>
    <Col
      className="policy-details-cover__title--label px-0"
      lg={8}
      md={6}
      xs={6}
    >
      <HDLabelRefactor
        Tag="h5"
        adjustImagePosition={true}
        className="mb-0"
        text="Comprehensive"
      />
    </Col>
  </div>
  <div
    className="policy-details-cover__list"
  >
    <HDTickListItem
      key="Motor Legal Expenses"
      selected={true}
      title="Motor Legal Expenses"
    />
    <HDTickListItem
      key="Roadside Assistance Breakdown cover"
      selected={true}
      title="Roadside Assistance Breakdown cover"
    />
    <HDTickListItem
      key="Driving other cars"
      selected={true}
      title="Driving other cars"
    >
      Third party cover for the policyholder only, see your Certificate of Insurance for more information.
    </HDTickListItem>
    <HDTickListItem
      key="Courtesy car"
      selected={true}
      title="Courtesy car"
    >
      A small car will be provided while your car's being repaired by one of our approved repairers.
    </HDTickListItem>
    <HDTickListItem
      key="Windscreen or windows"
      selected={false}
      title="Windscreen or windows"
    />
    <HDTickListItem
      key="Fire and theft damage"
      selected={false}
      title="Fire and theft damage"
    />
  </div>
</div>
`;

exports[`<HDContainer /> render component with title 1`] = `
<div>
  <HDLabelRefactor
    Tag="h4"
    adjustImagePosition={true}
    className="policy-details-cover__extra-title"
    text="title"
  />
  <div
    className="policy-details-cover__list"
  >
    <HDTickListItem
      key="Motor Legal Expenses"
      selected={true}
      title="Motor Legal Expenses"
    />
    <HDTickListItem
      key="Roadside Assistance Breakdown cover"
      selected={true}
      title="Roadside Assistance Breakdown cover"
    />
    <HDTickListItem
      key="Driving other cars"
      selected={true}
      title="Driving other cars"
    >
      Third party cover for the policyholder only, see your Certificate of Insurance for more information.
    </HDTickListItem>
    <HDTickListItem
      key="Courtesy car"
      selected={true}
      title="Courtesy car"
    >
      A small car will be provided while your car's being repaired by one of our approved repairers.
    </HDTickListItem>
  </div>
</div>
`;

exports[`<HDContainer /> render component withuot header 1`] = `
<div>
  <HDLabelRefactor
    Tag="h4"
    adjustImagePosition={true}
    className="policy-details-cover__extra-title"
    text="title"
  />
  <div
    className="policy-details-cover__list"
  >
    <HDTickListItem
      key="Motor Legal Expenses"
      selected={true}
      title="Motor Legal Expenses"
    />
    <HDTickListItem
      key="Roadside Assistance Breakdown cover"
      selected={true}
      title="Roadside Assistance Breakdown cover"
    />
    <HDTickListItem
      key="Driving other cars"
      selected={true}
      title="Driving other cars"
    >
      Third party cover for the policyholder only, see your Certificate of Insurance for more information.
    </HDTickListItem>
    <HDTickListItem
      key="Courtesy car"
      selected={true}
      title="Courtesy car"
    >
      A small car will be provided while your car's being repaired by one of our approved repairers.
    </HDTickListItem>
  </div>
</div>
`;
